src/service/resources-impl.js

```diff
@@ -1095,8 +1095,9 @@ export class Resources {
           // 3. Active elements are immediately resized. The assumption is that
           // the resize is triggered by the user action or soon after.
           resize = true;
-        } else if (topMarginDiff == 0 && box.bottom + Math.min(heightDiff, 0) >=
-              viewportRect.bottom - bottomOffset) {
+        } else if (box.top >= viewportRect.bottom - bottomOffset ||
```

Comment:
Good spot. You're right box.top does not include the margin. I've refined this so we calculate the top boundary that isn't changed and then check that this is below the fold.

With my refined approach, topMarginDiff being negative is not an issue, but I guess there is a potential issue if someone tries to set a top margin to be -100 or something. I've filed https://github.com/ampproject/amphtml/issues/8333 and will address this in a separate PR.